File: frontend/src/types/worker.ts
export type WorkerStatus = 'idle' | 'busy' | 'offline'

export interface Worker {
  id: string
  name: string
  status: WorkerStatus
  hostname: string
  pid: number
  max_concurrency: number
  active_jobs: number
  last_heartbeat_at: string | null
  created_at: string
}

export interface WorkerStats {
  worker_id: string
  worker_name: string
  hostname: string
  pid: number
  status: WorkerStatus
  active_jobs: number
  max_concurrency: number
  last_heartbeat_at: string | null
  completed_jobs: number
  failed_jobs: number
}

File: backend/app/schemas/worker.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.worker import WorkerStatus


class WorkerResponse(BaseModel):
    id: uuid.UUID
    name: str
    status: WorkerStatus
    hostname: str
    pid: int
    max_concurrency: int
    active_jobs: int
    last_heartbeat_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class WorkerStatsResponse(BaseModel):
    worker_id: uuid.UUID
    worker_name: str
    hostname: str
    status: WorkerStatus
    active_jobs: int
    max_concurrency: int
    last_heartbeat_at: datetime | None
    completed_jobs: int
    failed_jobs: int
